gui/src/components/onboarding/pages/body-proportions/HeightInput.tsx

```diff
@@ -0,0 +1,255 @@
+          'bg-background-50 hover:bg-background-40 cursor-pointer': !disabled,
+        }
+      )}
+      onClick={() => !disabled && onClick()}
+    >
+      <Typography
+        variant={isXs ? 'mobile-title' : 'section-title'}
+        color={disabled ? 'text-background-40' : 'primary'}
+      >
+        {format}
+      </Typography>
+      {unit == 'cm' && isXs && (
+        <Typography
+          id={`unit-${unit}`}
+          color={disabled ? 'text-background-40' : 'primary'}
+        />
+      )}
+    </div>
+  );
+}
+
+function UnitSelector({
+  name,
+  active,
+  onClick,
+}: {
+  name: string;
+  active: boolean;
+  onClick: () => void;
+}) {
+  return (
+    <div
+      className={classNames(
+        {
+          'bg-accent-background-30': active,
+          'hover:bg-background-40 bg-background-40': !active,
+        },
+        'flex items-center justify-center rounded-md outline-background-10 cursor-pointer'
+      )}
+      onClick={onClick}
+    >
+      <Typography id={name} />
+    </div>
+  );
+}
+
+function formatInFoot(meters: number, locale: string[]) {
+  const totalInches = Math.round(convert(meters, 'meter').to('inch'));
+  const feet = Math.floor(totalInches / 12);
+  const inches = totalInches % 12;
+
+  const feetFormatter = new Intl.NumberFormat(locale, {
+    style: 'unit',
+    unit: 'foot',
+    unitDisplay: 'narrow',
+    maximumFractionDigits: 0,
+  });
+
+  const inchFormatter = new Intl.NumberFormat(locale, {
+    style: 'unit',
+    unit: 'inch',
+    unitDisplay: 'narrow',
+    maximumFractionDigits: 0,
+  });
+
+  return `${feetFormatter.format(feet)} ${inchFormatter.format(inches)}`;
+}
+
+const round4Digit = (value: number) => Math.round(value * 10000) / 10000;
+
+export function HeightSelectionInput({
+  disabled = false,
+  hmdHeight,
+  setHmdHeight,
+}: {
+  disabled?: boolean;
+  hmdHeight: number;
+  setHmdHeight: (height: number) => void;
+}) {
+  if (!hmdHeight) disabled = true;
+
+  const { isXs } = useBreakpoint('xs');
+  const [unit, setUnit] = useState<'meter' | 'foot'>('meter');
+  const { currentLocales } = useLocaleConfig();
+
+  const formattedHeight = useMemo(() => {
+    if (!hmdHeight) return '--';
+
+    const fullHeight = hmdHeight / EYE_HEIGHT_TO_HEIGHT_RATIO;
+    const displayHeight = round4Digit(fullHeight);
+
+    if (unit === 'meter') {
+      return new Intl.NumberFormat(currentLocales, {
+        style: 'unit',
+        unit: 'meter',
+        maximumFractionDigits: 2,
+        minimumFractionDigits: 2,
+      }).format(displayHeight);
+    }
+
+    return formatInFoot(displayHeight, currentLocales);
+  }, [hmdHeight, unit]);
+
+  const incrementMath = (unit: 'inch' | 'cm' | 'foot', value: number) => {
+    const incrementInMeters = convert(value, unit).to('meter');
+    const oldFull = hmdHeight / EYE_HEIGHT_TO_HEIGHT_RATIO;
+    const newFull = oldFull + incrementInMeters;
+    const newEye = newFull * EYE_HEIGHT_TO_HEIGHT_RATIO;
```

Comment:
Incrementing in this way can cause unexpected behaviour to users due to the disaplayed value not matching the backend value, ex. "6 ft. 0 in." would be assumed to be about 182.88 cm, but if there was 1 cm added before switching to feet and inches, that 1 cm may persist (ex. 183.88 cm) while still displaying "6 ft. 0 in.". This isn't a massive issue since it's at most a matter of a few cm, but it's something that the user will not be able to control.